osmdroid-android/src/main/java/org/osmdroid/tileprovider/MapTileProviderBasic.java

```diff
@@ -41,13 +41,13 @@
 public class MapTileProviderBasic extends MapTileProviderArray implements IMapTileProviderCallback {
 
 	protected IFilesystemCache tileWriter;
-	private final INetworkAvailablityCheck mNetworkAvailabilityCheck;
+	private INetworkAvailablityCheck mNetworkAvailabilityCheck;
 
 	/**
 	 * @since 6.1.0
 	 */
-	private final MapTileDownloader mDownloaderProvider;
-	private final MapTileApproximater mApproximationProvider;
+	private MapTileDownloader mDownloaderProvider;
+	private MapTileApproximater mApproximationProvider;
```

Comment:
There's no obvious reason not to keep them `final`.
The point of this PR is to create these fields differently, not to make it possible to change them afterwards, isn't it?